components/clarinet-sdk/src/utils.rs

```diff
@@ -0,0 +1,222 @@
+        },
+        StacksTransactionEvent::FTEvent(FTEventType::FTBurnEvent(data)) => StacksEvent {
+            event: "ft_burn_event".into(),
+            data: data.json_serialize(),
+        },
+    }
+}
+
+pub fn to_raw_value(value: &Value) -> String {
+    let mut bytes = vec![];
+
+    value.consensus_serialize(&mut bytes).unwrap();
+    let raw_value = bytes
+        .iter()
+        .map(|b| format!("{:02x}", b))
+        .collect::<Vec<String>>();
+    format!("0x{}", raw_value.join(""))
+}
+
+pub fn uint8_to_string(mut value: &[u8]) -> String {
+    let value = Value::consensus_deserialize(&mut value).expect("failed to parse clarity value");
+    value_to_string(&value)
+}
+
+pub fn uint8_to_value(mut value: &[u8]) -> Value {
+    Value::consensus_deserialize(&mut value).expect("failed to parse clarity value")
+}
+
+fn value_to_string(value: &Value) -> String {
+    match value {
+        Value::Tuple(tup_data) => {
+            let mut data = Vec::new();
+            for (name, value) in tup_data.data_map.iter() {
+                data.push(format!("{}: {}", &**name, value_to_string(value)))
+            }
+            format!("{{ {} }}", data.join(", "))
+        }
+        Value::Optional(opt_data) => match opt_data.data {
+            Some(ref x) => format!("(some {})", value_to_string(x)),
+            None => "none".to_string(),
+        },
+        Value::Response(res_data) => match res_data.committed {
+            true => format!("(ok {})", value_to_string(&res_data.data)),
+            false => format!("(err {})", value_to_string(&res_data.data)),
+        },
+        Value::Sequence(SequenceData::String(CharType::ASCII(data))) => {
+            format!("\"{}\"", String::from_utf8(data.data.clone()).unwrap())
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(data))) => {
+            let mut result = String::new();
+            for c in data.data.iter() {
+                if c.len() > 1 {
+                    // escape extended charset
+                    result.push_str(&format!("\\u{{{}}}", hash::to_hex(&c[..])));
+                } else {
+                    result.push(c[0] as char)
+                }
+            }
+            format!("u\"{}\"", result)
+        }
+        Value::Sequence(SequenceData::List(list_data)) => {
+            let mut data = Vec::new();
+            for value in list_data.data.iter() {
+                data.push(value_to_string(value))
+            }
+            format!("(list {})", data.join(" "))
+        }
+        _ => format!("{}", value),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::value_to_string;
+    use clarity_repl::clarity::vm::types::{
+        ASCIIData, CharType, ListData, ListTypeData, OptionalData, ResponseData, SequenceData,
+        SequencedValue, TupleData, TypeSignature, UTF8Data, NONE,
+    };
+    use clarity_repl::clarity::vm::{ClarityName, Value};
+    use std::convert::TryFrom;
+
+    #[test]
+    fn test_value_to_string() {
```

Comment:
Definitely not crucial, but I've been loving the [test-case](https://github.com/frondeus/test-case) crate for this sort of thing. You can reuse the same test code, and generate multiple test cases that are run against the test. This way, if the first case fails, the rest are still run and it doesn't look like everything is broken. For this test, it could look something like this:

```rust

    #[test_case(Value::Int(42) => is equal_to  "42".to_string(); "for Int type")]
    #[test_case(Value::UInt(12345678909876) => is equal_to  "u12345678909876".to_string(); "for UInt type")]
    #[test_case(Value::Bool(true) => is equal_to  "true".to_string(); "for Bool type")]
    #[test_case(Value::buff_from(vec![1, 2, 3]).unwrap() => is equal_to  "0x010203".to_string(); "for buff type")]
    fn test_value_to_string(val: Value) -> String {
        value_to_string(&val)
    }
```

Reply:
Indeed this looks great! I'll give it a try in the next phases of this project